pubmed_parser/pubmed_oa_parser.py

```diff
@@ -87,6 +87,18 @@ def parse_article_meta(tree):
     return dict_article_meta
 
 
+def parse_date(tree, date_type):
+    """Parse ppub and epub dates."""
+    pub_year_node = tree.find(f".//pub-date[@pub-type='{date_type}']/year")
+    pub_year = pub_year_node.text if pub_year_node is not None else ""
```

Comment:
`pub_year_node = tree.find(f".//pub-date[@pub-type='{date_type}']/year") or ""`

and same for below

Reply:
You cannot evaluate a `lxml.etree.Element`. But I refactored the code with a helper function `get_text_or_default()` to fetch the text content of a node or return a default value if the node is None.